### PR TITLE
add validation stringency to bam parsing, flagstat

### LINKED ISSUE #973: Add stringency parameter to flagstat

### PR DESCRIPTION
**Pull request by ryan-williams**
fixes #973

### CODE REVIEW

File: adam-cli/src/main/scala/org/bdgenomics/adam/cli/FlagStat.scala [hunk 1]
```diff
@@ -41,6 +41,8 @@ class FlagStatArgs extends Args4jBase with ParquetArgs {
   val inputPath: String = null
   @Argument(required = false, metaVar = "OUTPUT", usage = "Optionally write the stats to this file.", index = 1)
   val outputPath: String = null
+  @Args4jOption(required = false, name = "-stringency", usage = "Set the parsing stringency: SILENT, LENIENT, STRICT.")
+  val stringency: String = "SILENT"
```
Comment (heuermh) [inline adam-cli/src/main/scala/org/bdgenomics/adam/cli/FlagStat.scala]: what is the default if SAMHeaderReader.VALIDATION_STRINGENCY_PROPERTY is not set?

Comment (ryan-williams) [inline adam-cli/src/main/scala/org/bdgenomics/adam/cli/FlagStat.scala]: [STRICT](https://github.com/samtools/htsjdk/blob/1.139/src/java/htsjdk/samtools/ValidationStringency.java#L20). `flagstat` is a simple enough operation that i thought `SILENT` was more often than not what a user would want.

Comment (heuermh) [inline adam-cli/src/main/scala/org/bdgenomics/adam/cli/FlagStat.scala]: ok, I agree with you on flagstat defaulting to SILENT, and I see the default in loadBam is STRICT.  +1

**Comment by heuermh** [comment]
Thanks @ryan-williams!

**Pull request merged**